api/tests/src/task/notifications/test_opportunity_notification.py

```diff
@@ -184,6 +185,9 @@ def set_env_var_for_email_notification_config(self, monkeypatch):
         monkeypatch.setenv("AWS_PINPOINT_APP_ID", "test-app-id")
         monkeypatch.setenv("FRONTEND_BASE_URL", "http://testhost:3000")
 
+        self.notification_config = EmailNotificationConfig()
+        self.notification_config.reset_emails_without_sending = False
```

Comment:
looking at this, depending on how Pytest builds the fixture dependency tree, this can be null when used below.

Make this a fixture that gets pulled in (or just define it in the notification_task fixture), relying on the side effect of another fixture without anything to enforce the ordering is going to cause issues potentially if there is some randomness to the fixture order resolution